### DISCUSSION TITLE
Using tags for data driven style

### DISCUSSION THREAD

**Discussion by hugosoli** [heremaps/harp.gl#2328] [Q&A]
Hi I want to have a Theme that has some data drive style. I upload a geojson with the Data Hub CLI and it works fine. One tag was authomatically generated from the file name

id                      │ geometry.type           │ **tags**                    │ createdAt               │ updatedAt               │
├─────────────────────────┼─────────────────────────┼─────────────────────────┼─────────────────────────┼─────────────────────────┤
│ p7RjANTPCd776Wez        │ Polygon                 │ **coloniascdmx**            │ 10/24/2021, 06:46 AM    │ 10/24/2021, 06:46 AM   

I have tries several option at the style description. for example:

{//colonias
          description: "GeoJson polygon",
          when: [
   "all",
   ["==", ["geometry-type"], "Polygon"],
   **["==", ["get", "@ns:com:here:xyz.tags[0]"], "coloniascdmx"],**
],
          technique: "fill",
          attr: {
            color: "#00000010",
            lineColor: "#FF000010",
            lineWidth: 1
          }
        }

The object sees to have such feature:

userData:
$geometryType: "polygon"
$id: "yaMmk6oT6U4MECC3"
$layer: "NUcQuJnH"
$level: 13
$zoom: 14
**@ns:com:here:xyz: "{\"tags\": [\"coloniascdmx\"], \"space\": \"NUcQuJnH\", \"createdAt\": 1635075994241, \"updatedAt\": 1635075994241}"**
alcaldi: "CUAUHTEMOC"
cve_alc: 15
cve_col: "15-011"
entidad: 9
g_pnt_2: "19.4204440063,-99.1282884556"
geo_shp: "{\"type\": \"Polygon\", \"coordinates\": [[[-99.12609109908855, 19.42281743313811], [-99.12658750563517, 19.421710953869873], [-99.12713303937066, 19.420494952326166], [-99.12716249417296, 19.42030031448885], [-99.12735449833221, 19.41903153638596], [-99.12740"
id: "yaMmk6oT6U4MECC3"
nombre: "ESPERANZA"
secc_cm: "4892"
secc_pr: "4893"
_id: 1732

Any idea or help?
Cheers and thanks
Hugo

**Comment by nzjony** [comment]
@hugosoli , does something like: `["in", ["get", "tags", ["get", "xyz"]], "coloniascdmx"]` work?

You can use the get syntax to return an object, which you can then query for the `tags` property, see: https://github.com/heremaps/harp.gl/blob/master/@here/harp-datasource-protocol/StyleExpressions.md#get

**Reply by hugosoli** [reply]
Thanks for the rely.
Unfortunately I had no luck. The previous answer help me to dig more but still not sure how to do it. I have read all the StyleExpression documentation but no luck.
//this would be the ideal logic. which is check if you have a tag with the name "estaciones_del_metro". If so draw the cirlce.

["in", "estaciones_del_metro", ["get", "tags"]]

//I have also tried

when: ["==", ["in", "estaciones_del_metro", ["get", "tags"]], true]

the problem with the previos awner (may be I did not explained to me properly is that I want ALL the elements of certain tag.

["==", ["get", "tags", ["get", "xyz"]], "estaciones_del_metro"]

this seems to require "xyz" something. is "xyz" an example or is the real name where the object is stored (@ns:com:here:xyz ??


For example 

when: ["in", ["literal","estaciones_del_metro"],["get", "tags", ["get", "xyz"]]],

ConsoleChannel.js:23 WorkerChannel: StyleSetEvaluator: failed to evaluate expression ["in",["literal","estaciones_del_metro"],["get","tags",["get","xyz"]]] error Error: expected an object or array literal

the last one paints all the objects for example.

I am near but still missing something. :-(

**Reply by nzjony** [reply]
I had my syntax wrong, try: `[
                                "in",
                                "coloniascdmx",
                                ["get", "tags", ["get", "@ns:com:here:xyz"]]
                            ]`
                            
See also: https://github.com/heremaps/harp.gl/commit/5791fd485c9e4c682f67b40a130995e1c9deff8c

**Reply by hugosoli** [reply]
Thanks for the reply. Unfortunately is not working :(

          when: ["in", "estaciones_del_metro", ["get", "tags", ["get", "@ns:com:here:xyz"]]],
is not drawing. 
when: ["in", "areas_verdes", ["get", "tags", ["get", "@ns:com:here:xyz"]]],
not working, 
when: ["in", "coloniascdmx", ["get", "tags", ["get", "@ns:com:here:xyz"]]],
not working, 
I am sure I have those tags on my database.

id                      │ geometry.type           │ tags                    │ createdAt               │ updatedAt      
│ 6gC9FJxEVvi41zlo        │ Point                   │ estaciones_del_metro    │ 10/26/2021, 02:56 PM    │ 10/26/2021, 02:56 PM 
│ G5dIXEy3DU1fhCvO        │ Point                   │ areas_verdes            │ 10/26/2021, 02:57 PM    │ 10/26/2021, 02:57 PM  
G4b7GfAu5T8Hph0c        │ Polygon                 │ coloniascdmx            │ 10/26/2021, 02:55 PM    │ 10/26/2021, 02:55 PM  

if I run 
here xyz show IjL4qOrm -v
in CLI
I map appears and the tags are there.
![Screenshot from 2021-10-27 22-12-07](https://user-images.githubusercontent.com/3760099/139181035-1757380b-0e8a-4230-ac83-f08a7aaf2f31.png)

**Reply by nzjony** [reply]
@hugosoli , can you send an example of the geojson?

**Reply by hugosoli** [reply]
OF course, what I am doing is populating  with
here xyz upload -f estaciones_del_metro.json
with that I get the ID and then
here xyz upload IjL4qOrm -f mercados.json
The tag of the filename is created for each file. I am attaching both files
[data.zip](https://github.com/heremaps/harp.gl/files/7434222/data.zip)
.

**Reply by nzjony** [reply]
Ok, what does the geojson look like after you have uploaded it to here? Do you have some sort of sample? I was hoping to have have the date with the tags like you mentioned, so I could try it locally. Is that possible?

**Reply by nzjony** [reply]
I'm away Friday, so maybe I can only take a look on Monday sorry.

**Reply by hugosoli** [reply]
Hi, sorry for bothering. I am still fighting with this issue. Any recommendation?
Cheers

**Reply by nzjony** [reply]
@hugosoli , can you download the geojson from the xyz and see how it is stored?

**Reply by hugosoli** [reply]
hi!
I don't know exactly how to download the geojson but I guess It can be done with the CLI as follow:

>>here xyz show IjL4qOrm -r > alltogether.geojson
with that I get a gejson that has all the points with the properties. One element for example is:

{
      "id": "Lo2giU3wakmaI9Gm",
      "type": "Feature",
      "properties": {
        "id": 1221,
        "categoria": "Plazas y jardines",
        "subcategoria": "Jardines públicos",
        **"@ns:com:here:xyz":** {
          "space": "IjL4qOrm",
          "createdAt": 1635278264923,
          "updatedAt": 1635278264923
        }
      },
      "geometry": {
        "type": "Point",
        "coordinates": [
          -99.10662628,
          19.31315855,
          0
        ]
      }
    },

INTERESTING! There is nothing related with the tags on it. That could be the reason why I can not get it, howerver, the tags values should be somehwere because they apear on the userdata once it is downloaded

userData:
$geometryType: "polygon"
$id: "yaMmk6oT6U4MECC3"
$layer: "NUcQuJnH"
$level: 13
$zoom: 14
**@ns:com:here:xyz: "{"tags": ["coloniascdmx"], "space": "NUcQuJnH", "createdAt": 1635075994241, "updatedAt": 1635075994241}"**
alcaldi: "CUAUHTEMOC"
cve_alc: 15
cve_col: "15-011"
entidad: 9
g_pnt_2: "19.4204440063,-99.1282884556"
geo_shp: "{"type": "Polygon", "coordinates": [[[-99.12609109908855, 19.42281743313811], [-99.12658750563517, 19.421710953869873], [-99.12713303937066, 19.420494952326166], [-99.12716249417296, 19.42030031448885], [-99.12735449833221, 19.41903153638596], [-99.12740"
id: "yaMmk6oT6U4MECC3"
nombre: "ESPERANZA"
secc_cm: "4892"
secc_pr: "4893"
_id: 173
[alltogether.zip](https://github.com/heremaps/harp.gl/files/7475777/alltogether.zip)
2

here the link of the zip of the downloaded file.
Any ideas?
cheers
